src/instructlab/sdg/generate_data.py

```diff
@@ -210,15 +210,28 @@ def _context_init(
         **extra_kwargs,
     )
 
-
 def _sdg_init(ctx, pipeline):
     pipeline_pkg = None
 
     # Search for the pipeline in User and Site data directories
     # then for a package defined pipeline
     # and finally pipelines referenced by absolute path
-    data_dirs = [os.path.join(xdg_data_home(), "instructlab", "sdg")]
-    data_dirs.extend(os.path.join(dir, "instructlab", "sdg") for dir in xdg_data_dirs())
+    data_dir = os.path.join(xdg_data_home(), "instructlab", "sdg", "models", "docling")
+    data_dirs = [data_dir]
+    data_dirs.extend(
+        os.path.join(dir, "instructlab", "sdg", "models", "docling") for dir in xdg_data_dirs()
+    )
+
+    # Set `docling_model_path` to consistently use `data_dir`
+    docling_model_path = Path(data_dir)
+    os.makedirs(docling_model_path, exist_ok=True)
+
+    if not os.listdir(docling_model_path):
+        # Download models if directory is empty
+        logger.info("Docling models for chunking not found locally. Downloading from Hugging Face...")
+        StandardPdfPipeline.download_models_hf()
```

Comment:
I would suggest moving this as close to where docling code gets executed as possible - rather than right out of the gate here

Reply:
addressed